gcloud/pubsub/topic.py

```diff
@@ -113,6 +122,8 @@ def publish(self, message, **attrs):
         :rtype: str
         :returns: message ID assigned by the server to the published message
         """
+        if self.add_timestamp_to_messages:
+            attrs['timestamp'] = '%sZ' % _NOW().isoformat()
```

Comment:
I disagree:  overriding a default at a more local call point is very common (in fact, we do it ourselves with connection / project).  It isn't necessarily a programming error on the user's part:  "consenting adults", etc.